File: Experiment/ServerBase/sensorsimulator/src/main/java/com/swot/sensorsimulator/example/SensorLocation.java
package com.swot.sensorsimulator.example;

public class SensorLocation {
    private final double latitude;
    private final double longitude;
    private final String zone;

    public SensorLocation(double latitude, double longitude, String zone) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.zone = zone;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public String getZone() {
        return zone;
    }
}

File: Resources/python/sensors/src/models/sensor_location.py
from dataclasses import dataclass


@dataclass(frozen=True)
class SensorLocation:
    """Immutable data class representing a sensor's physical location."""
    latitude: float
    longitude: float
    zone: str


# interfaces/message_publisher.py
from abc import ABC, abstractmethod
from typing import Any


class MessagePublisher(ABC):
    """Abstract base class defining the interface for message publishing."""

    @abstractmethod
    def connect(self) -> None:
        """Establish connection to the message broker."""
        pass

    @abstractmethod
    def publish(self, topic: str, message: Any, qos: int = 0) -> None:
        """Publish a message to the specified topic."""
        pass

    @abstractmethod
    def disconnect(self) -> None:
        """Disconnect from the message broker."""
        pass